### PR TITLE
Additional MD and Boundary Condition  entities

### PR DESCRIPTION
**Pull request by ahashibon**
@mehdisadeghi please check the additions to the MD and boundary conditions.

### CODE REVIEW

File: yaml_files/simphony_metadata.yml [hunk 1]
```diff
@@ -74,6 +74,8 @@ CUDS_KEYS:
   BOUNDARY:
     definition: System boundary
     parent: CUBA.CUDS_COMPONENT
+    CUBA.CONDITION:
+      shape (:)
 
   CONDITION:
     definition: Condition on boundaries or model entities
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml]: CONDITION is missing parent attribute

File: yaml_files/simphony_metadata.yml [hunk 2]
```diff
@@ -74,6 +74,8 @@ CUDS_KEYS:
   BOUNDARY:
     definition: System boundary
     parent: CUBA.CUDS_COMPONENT
+    CUBA.CONDITION:
+      shape (:)
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml]: A colon is missing after `shape` attribute.

File: yaml_files/simphony_metadata.yml [hunk 3]
```diff
@@ -263,16 +300,16 @@ CUDS_KEYS:
     CUBA.MATERIAL:
       shape: (2)
 
-  DIRICHLET_BOUNDARY_CONDITION:
+  DIRICHLET:
     definition: Dirichlet boundary condition
     parent: CUBA.CONDITION
     models: [CUBA.CONTINUUM]
-    variables:
+    CUBA.VARIABLES:
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml]: `CUBA.VARIABLES` is undefined.

File: yaml_files/simphony_metadata.yml [hunk 4]
```diff
@@ -281,7 +318,7 @@ CUDS_KEYS:
     CUBA.MATERIAL:
       shape: (:)
 
-  EMPTY_BOUNDARY_CONDITION:
+  EMPTY:
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml:326]: `EMPTY` is very generic term, could it be more specific?

File: yaml_files/simphony_metadata.yml [hunk 5]
```diff
@@ -290,6 +327,16 @@ CUDS_KEYS:
     CUBA.MATERIAL:
       shape: (:)
 
+  FREE:
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml:337]: Same comment as `EMPTY` here.

File: yaml_files/simphony_metadata.yml [hunk 6]
```diff
@@ -106,6 +108,41 @@ CUDS_KEYS:
...
+    CUBA.COUPLING_TIME:
+      default: 1.0
+
+  BERENDSEN: 
+    definition: The Berendsen thermostat model for temperature rescaling of all particles each time step. The coupling time specifies how rapidly the temperature should be relaxed or coupled to the bath. 
+    parent: CUBA.THERMOSTAT
+    models: [CUBA.ATOMISTIC, CUBA.MESOSCOPIC]
+    CUBA.TEMPERATURE:
+      shape: (2)
+      default: (0.0, 0.0)
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml]: If this is an array, use rectangular brackets.

File: yaml_files/simphony_metadata.yml [hunk 7]
```diff
@@ -106,6 +108,41 @@ CUDS_KEYS:
...
+    models: [CUBA.ATOMISTIC, CUBA.MESOSCOPIC]
+
+  NOSE_HOOVER: 
+    definition: The Nose-Hoover thermostat: add an extra term to the equation of motion to model the interaction with a heat bath. The coupling time specifies how rapidly the temperature should be coupled to the bath. 
+    parent: CUBA.THERMOSTAT
+    models: [CUBA.ATOMISTIC, CUBA.MESOSCOPIC]
+    CUBA.TEMPERATURE:
+      shape: (2)
+      default: (0.0, 0.0)
+    CUBA.COUPLING_TIME:
```
Comment (mehdisadeghi) [inline yaml_files/simphony_metadata.yml:124]: `COUPLING_TIME` is not defined.

**Comment by mehdisadeghi** [comment]
@ahashibon This is ready to merge.

**Comment by kitchoi** [comment]
@mehdisadeghi This branch is based on is a fairly old master.  Could you merge master to this branch to make sure PR tests pass please?

**Comment by mehdisadeghi** [comment]
@kitchoi good point, merged it.

File: yaml_files/simphony_metadata.yml [hunk 8]
```diff
@@ -438,9 +513,12 @@ CUDS_KEYS:
       scope: CUBA.USER
 
   BOX:
-    definition: A simple hexahedron box object
+    definition: A simple hexahedron (with six faces) simulation box defined by the three vectors and three directions. The condition should be specified for each direction (two faces at a time).
```
Comment (stefanoborini) [inline yaml_files/simphony_metadata.yml:516]: When defining 3x3 matrices representing three vectors, it's also useful to be clear if they are along the rows or the columns. I've seen so many errors due to this that it's never bad to overspecify.

Comment (ahashibon) [reply yaml_files/simphony_metadata.yml:516]: we should aim to be consistent and if 3 vectors are represented by a matrix, then it should be row wise always! 
changed the comment in the file.

**Pull request merged**